tests/core/test_simulations.py

```diff
@@ -1,20 +1,72 @@
 # -*- coding: utf-8 -*-
 
 from __future__ import unicode_literals, print_function, division, absolute_import
-from openfisca_country_template.situation_examples import single
+import os
 
 from openfisca_core.simulations import Simulation
+from openfisca_core.variables import Variable
+from openfisca_core.parameters import ParameterNode
+from openfisca_core.periods import MONTH
+
+from openfisca_country_template.situation_examples import single
+from openfisca_country_template.entities import Person
 
 from .test_countries import tax_benefit_system
 
+
+class zone(Variable):
+    value_type = str
+    entity = Person
+    definition_period = MONTH
+    label = "Variable for fancy indexing inputs"
+
+
+class fancy_indexing(Variable):
+    value_type = int
+    entity = Person
+    definition_period = MONTH
+    label = "Variable using fancy indexing"
+
+    def formula(person, period, parameters):
+        zone = person('zone', period)
+        return parameters(period).rate.single.owner[zone]
+
+
+LOCAL_DIR = os.path.dirname(os.path.abspath(__file__))
+fancy_indexing_directory = os.path.join(LOCAL_DIR, 'parameters_fancy_indexing')
+parameters = ParameterNode(directory_path = fancy_indexing_directory)
+tax_benefit_system.parameters.merge(parameters)
+
+tax_benefit_system.add_variable(fancy_indexing)
+tax_benefit_system.add_variable(zone)
+
 scenario = tax_benefit_system.new_scenario().init_from_attributes(
-    period=2014
+    period='2017-01', input_variables={'zone': ['z1', 'z1', 'z1']}
     )
 
 
 def test_calculate_with_trace():
     simulation = scenario.new_simulation(trace=True)
-    simulation.calculate('disposable_income', "2014-01")
+    simulation.calculate('income_tax', '2017-01')
+
+    salary_trace = simulation.tracer.trace['salary<2017-01>']
+    assert salary_trace['parameters'] == {}
+
+    income_tax_trace = simulation.tracer.trace['income_tax<2017-01>']
+    assert income_tax_trace['parameters']['taxes.income_tax_rate<2017-01-01>'] == 0.15
+
+    # Trace parameters called with indirect access
+    simulation.calculate('housing_tax', '2017')
+    housing_tax_trace = simulation.tracer.trace['housing_tax<2017>']
+    assert 'taxes.housing_tax<2017-01-01>' not in housing_tax_trace['parameters']
+    assert housing_tax_trace['parameters']['taxes.housing_tax.rate<2017-01-01>'] == 10
+    assert housing_tax_trace['parameters']['taxes.housing_tax.minimal_amount<2017-01-01>'] == 200
+
+    # Tracing of fancy indexed parameters is not yet supported
```

Comment:
If we were to test the tracer with fancy indexing, I think it would be easier to write a unit test on the tracer instead of mutating the test tax benefit system, which causes side-effects (cf. `__pycache__` bug you reported).

But we have not yet implemented it, so I don't really see the point of a test that check that we don't do something we are going to implement later.

Reply:
The point is that if we hadn't written this test we wouldn't have noticed that the tracing as originally *breaks* fancy indexing; what this unit test specifies is that fancy indexing still works in a simulation with trace on.